cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -340,3 +399,277 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration, volumes []v1.Vo
+	if err := createTLSSecretFromFiles(
+		kubeadmconstants.CACertAndKeyBaseName,
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.CACertName),
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.CAKeyName),
+		client); err != nil {
+		return err
+	}
+	if err := createTLSSecretFromFiles(
+		kubeadmconstants.APIServerCertAndKeyBaseName,
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.APIServerCertName),
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.APIServerKeyName),
+		client); err != nil {
+		return err
+	}
+	if err := createTLSSecretFromFiles(
+		kubeadmconstants.APIServerKubeletClientCertAndKeyBaseName,
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.APIServerKubeletClientCertName),
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.APIServerKubeletClientKeyName),
+		client); err != nil {
+		return err
+	}
+	if err := createTLSSecretFromFiles(
+		kubeadmconstants.ServiceAccountKeyBaseName,
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.ServiceAccountPublicKeyName),
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.ServiceAccountPrivateKeyName),
+		client); err != nil {
+		return err
+	}
+	if err := createTLSSecretFromFiles(kubeadmconstants.FrontProxyCACertAndKeyBaseName,
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.FrontProxyCACertName),
+		path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "pki", kubeadmconstants.FrontProxyCAKeyName),
+		client); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createControllerManagerKubeconfigSecret(client *clientset.Clientset) error {
+	file := path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, kubeadmconstants.ControllerManagerKubeConfigFileName)
+	if err := createOpaqueSecretFromFile(kubeadmconstants.ControllerManagerKubeConfigFileName, file, client); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createSchedulerKubeconfigSecret(client *clientset.Clientset) error {
+	file := path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, kubeadmconstants.SchedulerKubeConfigFileName)
+	if err := createOpaqueSecretFromFile(kubeadmconstants.SchedulerKubeConfigFileName, file, client); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createTLSSecretFromFiles(secretName, crt, key string, client *clientset.Clientset) error {
+	data := make(map[string][]byte, 0)
+
+	crtBytes, err := ioutil.ReadFile(crt)
+	if err != nil {
+		return err
+	}
+	data["tls.crt"] = crtBytes
+
+	keyBytes, err := ioutil.ReadFile(key)
+	if err != nil {
+		return err
+	}
+	data["tls.key"] = keyBytes
+
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: metav1.NamespaceSystem,
+		},
+		Type: v1.SecretTypeTLS,
+		Data: data,
+	}
+
+	if _, err := client.Secrets(metav1.NamespaceSystem).Create(secret); err != nil {
```

Comment:
I think it would be better if the following functions returned a Secret object instead to make it unit-testable